pkg/networkservice/common/heal/context.go

```diff
@@ -18,23 +18,32 @@ package heal
 
 import (
 	"context"
+
+	"github.com/networkservicemesh/api/pkg/api/networkservice"
 )
 
 const (
-	registerClientFuncKey contextKeyType = "RegisterFunc"
+	requestHealFuncKey contextKeyType = "requestHealFuncKey"
 )
 
 type contextKeyType string
 
-func withRegisterClientFunc(parent context.Context, registerClientFunc RegisterClientFunc) context.Context {
+// requestHealFuncType - function used to inform heal server that connection should be healed
+//                       - conn              - connection to be healed
+//                       - restoreConnection - flag to specify whether what to heal.
+//                                             When false, only heals this particular connection.
+//                                             When true also tries to heal physical connection to next hop.
+type requestHealFuncType func(conn *networkservice.Connection, restoreConnection bool)
```

Comment:
It looks like we actually have two functions that hid by the boolean flag. Please consider removing the boolean flag and create two separate functions:
1. onRestoreConnection(conn *networkservice.Connection)
2. onHealConnection(conn *networkservice.Connection)